src/sea.ts

```diff
@@ -0,0 +1,253 @@
+  checkCompatibility();
+
+  const requiredOptions = await getOptions(options);
+  await createFiles(requiredOptions);
+  await createBlob(requiredOptions);
+  await createBinary(requiredOptions);
+  await createSeaReceiver(requiredOptions);
+  await cleanup(requiredOptions);
+
+  return requiredOptions;
+}
+
+async function createSeaReceiver(options: InternalSeaOptions) {
+  const receiverPath = path.join(options.dir, FILENAMES.SEA_RECEIVER);
+
+  await fs.ensureFile(receiverPath);
+  await fs.writeFile(receiverPath, SEA_RECEIVER_SCRIPT);
+}
+
+async function createFiles(options: InternalSeaOptions) {
+  const { dir, bin } = options;
+  const receiverPath = path.join(options.dir, FILENAMES.SEA_RECEIVER);
+
+  // sea-config.json
+  await fs.outputJSON(path.join(dir, FILENAMES.SEA_CONFIG), {
+    main: FILENAMES.SEA_MAIN,
+    output: FILENAMES.SEA_BLOB,
+    disableExperimentalSEAWarning: true
+  }, {
+    spaces: 2
+  });
+
+  // signtool.js
+  const binPath = bin || process.execPath;
+  const script = SEA_MAIN_SCRIPT
+    .replace('%PATH_TO_BIN%', escapeMaybe(binPath))
+    .replace('%PATH_TO_SCRIPT%', escapeMaybe(receiverPath))
+    .replace('%WINDOWS_SIGN_OPTIONS%', JSON.stringify(options.windowsSign));
+
+  await fs.outputFile(path.join(dir, FILENAMES.SEA_MAIN), script);
+}
+
+async function createBlob(options: InternalSeaOptions) {
+  const args = ['--experimental-sea-config', 'sea-config.json'];
+  const bin = process.execPath;
+  const cwd = options.dir;
+
+  log(`Calling ${bin} with options:`, args);
+
+  return spawnPromise(bin, args, {
+    cwd
+  });
+}
+
+async function createBinary(options: InternalSeaOptions) {
+  const { dir, filename } = options;
+
+  log(`Creating ${filename} in ${dir}`);
+
+  // Copy Node over
+  const seaPath = path.join(dir, filename);
+  await fs.copyFile(process.execPath, seaPath);
+
+  // Remove the Node signature
+  const signtool = path.join(__dirname, '../../vendor/signtool.exe');
+  await spawnPromise(signtool, [
+    'remove',
+    '/s',
+    seaPath
+  ]);
+
+  // Inject the blob
+  const blob = await fs.readFile(path.join(dir, FILENAMES.SEA_BLOB));
+  await postject.inject(seaPath, 'NODE_SEA_BLOB', blob, {
+    sentinelFuse: 'NODE_SEA_FUSE_fce680ab2cc467b6e072b8b5df1996b2'
+  });
+}
+
+async function cleanup(options: InternalSeaOptions) {
+  const { dir } = options;
+  const toRemove = [
+    FILENAMES.SEA_BLOB,
+    FILENAMES.SEA_MAIN,
+    FILENAMES.SEA_CONFIG
+  ];
+
+  for (const file of toRemove) {
+    try {
+      await fs.remove(path.join(dir, file));
+    } catch (error) {
+      console.warn(`Tried and failed to remove ${file}. Continuing.`, error);
+    }
+  }
+}
+
+async function getOptions(options: Partial<SeaOptions>): Promise<InternalSeaOptions> {
+  const cloned = { ...options };
+
+  if (!cloned.path) {
+    cloned.path = path.join(os.homedir(), '.electron', 'windows-sign', 'sea.exe');
+    await fs.mkdirp(cloned.path);

```

Comment:
Question: don't know if it makes a difference, but it looks like `fs.ensureFile` would be a better fit since it's the path to a file, not to a folder?

Reply:
Thank you!